### PR TITLE
Create support for concatenating semantic embeddings

### PR DESCRIPTION
**Pull request by siddhu001**
Added support for concatenating glove embeddings to SLU embeddings

### CODE REVIEW

**Comment by siddhu001** [comment]
Still generating results, not ready to merge.

**Comment by siddhu001** [comment]
This pull request is ready for review.

File: main.py [hunk 1]
```diff
@@ -56,26 +68,64 @@
...
 
 	# Initialize final model
-	model = Model(config=config)
+	if use_semantic_embeddings:
+		Sy_word = []
+		with open(os.path.join(config.folder, "pretraining", "words.txt"), "r") as f:
+			for line in f.readlines():
+				Sy_word.append(line.rstrip("\n"))
+		glove_embeddings=obtain_glove_embeddings(semantic_embeddings_path, Sy_word )
+		model = Model(config=config,pipeline=False, use_semantic_embeddings = use_semantic_embeddings, glove_embeddings=glove_embeddings)
```
Comment (viswavi) [inline main.py:82]: Very minor, but you can just have `use_semantic_embeddings = True` here and in the clause below.
Doesn't affect the accuracy of this code so you can just ignore this message

File: main.py [hunk 2]
```diff
@@ -104,42 +154,80 @@
 			Sy_word.append(line.rstrip("\n"))
 	train_dataset, valid_dataset, test_dataset = get_SLU_datasets(config)
 
+	if postprocess_words:
+		log_file="log_pipeline_postprocess.csv"
+	else:
+		if finetune_embedding:
+			log_file="log_pieline_finetune.csv"
```
Comment (viswavi) [inline main.py]: typo

File: main.py [hunk 3]
```diff
@@ -104,42 +154,80 @@
...
 
 if pipeline_gold_train:
 	# Generate datasets
-	train_dataset, valid_dataset, test_dataset = get_SLU_datasets(config,use_gold_utterances=True)
+	train_dataset, valid_dataset, test_dataset = get_SLU_datasets(config,use_gold_utterances=True,random_split=random_split, disjoint_split=disjoint_split)
 
 	# Initialize final model
-	model = Model(config=config,pipeline=True)
+	if use_semantic_embeddings:
+		glove_embeddings=obtain_glove_embeddings(semantic_embeddings_path, train_dataset.Sy_word )
```
Comment (viswavi) [inline main.py:212]: no fasttext option here?

File: models.py [hunk 4]
```diff
@@ -650,11 +653,69 @@ def infer(self, encoder_outputs, Sy, B=4, debug=False, y_lengths=None):
 
 		return beam_scores, beam
 
+def obtain_glove_embeddings(filename, vocab,dim=100):
```
Comment (viswavi) [inline models.py:656]: For future reference, there's a package called Gensim that makes it fairly easy to load any kind of word embedding file. I use it a lot

Comment (viswavi) [reply models.py:656]: But this works well too 👍 , just more work for you 😄

**Review by viswavi — APPROVED** [review]
This looks good. Left a few comments

File: main.py (hunk 3 above)
Comment (siddhu001) [reply main.py:212]: Since performance in end to end much greater than performance in pipeline, I decided to not further experiment with pipeline manner. Hence did not added fasttext option here.

File: main.py (hunk 1 above)
Comment (siddhu001) [reply main.py:82]: Makes sense. I think I will need to refactor code a bit later.
